### DISCUSSION TITLE
Server-side field validation error message is replaced with GraphQL query error in production mode

### DISCUSSION THREAD

**Discussion by Stalex89** [solspace/craft-freeform#718] [Questions & Answers] [answered]
I'm trying to implement Next.js frontend + Freeform in Headless CraftCMS, when I'm submitting form through the GraphQL mutation in development mode, I can see the array of validation errors per each field:

```
{
  "errors": [
    {
      "message": "[{\"firstName\":[\"This field is required\"]},{\"lastName\":[\"This field is required\"]},{\"companyName\":[\"This field is required\"]},{\"email\":[\"This field is required\"]}]",
      "extensions": {
        "category": "user"
      },
      "trace": [
        {
         ...
        }
     ]
  }
],
 "data": {
    "save_testForm_Submission": null
  }
```

However, if I switch Craft to production mode, I will get the generic GraphQL error message after the form is submitted:

```
{
  "errors": [
    {
      "message": "Something went wrong when processing the GraphQL query.",
      "extensions": {
        "category": "graphql"
      }
    }
  ],
  "data": {
    "save_famTripEn_Submission": null
  }
}
```

Can you tell me if there's a way to get server-side validation results/errors from Freeform through the GraphQL mutation response ?
Thank you in advance!

**Comment by kjmartens** [comment, marked as answer]
Sorry about this @Stalex89.

We've corrected the issue in Freeform 4.1.3 now. 🙂

**Reply by Stalex89** [reply]
It's working now, thank you very much for a quick fix!
